test/functional/wallet_resendwallettransactions.py

```diff
@@ -49,16 +49,21 @@ def run_test(self):
         block.solve()
         node.submitblock(ToHex(block))
 
-        # Transaction should not be rebroadcast
         node.syncwithvalidationinterfacequeue()
-        node.p2ps[1].sync_with_ping()
+        now = int(time.time())
```

Comment:
@MarcoFalke I removed the `sync_with_ping()` call because I think its a no-op, but I'd like your review to make sure I'm not (re)introducing a race in weird conditions. 

my understanding: 
`sync_with_ping()` is a p2p interface method that ensures any messages it has sent to the node have been processed by the node. 

the `syncwithvalidationinterfacequeue()` makes sense to me bc that's making sure the wallet gets notified about the block.

since we're trying to check if the wallet kicks off rebroadcasting the transaction, so this synchronization should not be relevant? it was definitely not sufficient. the test previously wasn't checking anything bc it didn't give enough time for a rebroadcast to kickoff, so I've added a `time.sleep()` call. 

tldr; I removed the `sync_with_ping()` bc I think it was a no-op & misleading. I've added a `time.sleep(2)` to actually test that a transaction isn't rebroadcast. please lmk if I'm missing something.

Reply:
>  sync_with_ping() is a p2p interface method that ensures any messages it has sent to the node have been processed by the node.

Correct. Also, any messages that the node has put in the send buffer (before the pong is sent) will get "flushed out" to the mininode.

Reply:
👍 . so, do you agree that seemed irrelevant here?